src/mpi/coll/algorithms/common/algo_common.h

```diff
@@ -16,13 +16,15 @@ static int MPII_Algo_compare_int(const void *a, const void *b)
 }
 
 /* Avoid unused function warning in certain configurations */
-static inline int MPIR_Algo_calculate_pipeline_chunk_info(int chunk_size, int type_size, int count,
-                                                          int *num_segments, int *segsize_floor,
-                                                          int *segsize_ceil) ATTRIBUTE((unused));
-static inline int MPIR_Algo_calculate_pipeline_chunk_info(int chunk_size,
-                                                          int type_size, int count,
-                                                          int *num_segments,
-                                                          int *segsize_floor, int *segsize_ceil)
+static inline int MPIR_Algo_calculate_pipeline_chunk_info(int chunk_size, int type_size,
```

Comment:
It probably makes sense to go ahead and promote chunk_size and type_size here, too. Unless there is some reason to keep them as int?